docs/docs/tutorials/deploying_and_interacting_with_smart_contracts.md

```diff
@@ -0,0 +1,382 @@
+
+const solc = require('solc');
+const path = require('path');
+const fs = require('fs');
+
+const fileName = 'MyContract.sol';
+const contractName = 'MyContract';
+
+// Read the Solidity source code from the file system
+const contractPath = path.join(__dirname, fileName);
+const sourceCode = fs.readFileSync(contractPath, 'utf8');
+
+// solc compiler config
+const input = {
+	language: 'Solidity',
+	sources: {
+		[fileName]: {
+			content: sourceCode,
+		},
+	},
+	settings: {
+		outputSelection: {
+			'*': {
+				'*': ['*'],
+			},
+		},
+	},
+};
+
+// Compile the Solidity code using solc
+const compiledCode = JSON.parse(solc.compile(JSON.stringify(input)));
+
+// Get the bytecode from the compiled contract
+const bytecode = compiledCode.contracts[fileName][contractName].evm.bytecode.object;
+
+// Write the bytecode to a new file
+const bytecodePath = path.join(__dirname, 'MyContractBytecode.bin');
+fs.writeFileSync(bytecodePath, bytecode);
+
+// Log the compiled contract code to the console
+console.log('Contract Bytecode:\n', bytecode);
+
+// Get the ABI from the compiled contract
+const abi = compiledCode.contracts[fileName][contractName].abi;
+
+// Write the Contract ABI to a new file
+const abiPath = path.join(__dirname, 'MyContractAbi.json');
+fs.writeFileSync(abiPath, JSON.stringify(abi, null, '\t'));
+
+// Log the Contract ABI to the console
+console.log('Contract ABI:\n', abi);
+```
+
+This code reads the Solidity code from the `MyContract.sol` file, compiles it using `solc`, and generates the ABI and bytecode for the smart contract. It then writes the bytecode to a new file called `MyContractBytecode.bin` and the contract ABI to `MyContractAbi.json`. And it logs them to the console.
+
+Run the following command to compile the Solidity code:
+
+```
+node compile.js
+```
+
+If everything is working correctly, you should see both the Contract Bytecode and the Contract ABI logged to the console.
+
+:::tip
+📝 There are couple of other ways to get the bytecode and the ABI like using Truffle development framework and running `truffle compile` (https://trufflesuite.com/docs/truffle/quickstart/#compile).
+Another way is to use Remix and check the _Compilation Details_ after compiling the smart contract (https://remix-ide.readthedocs.io/en/latest/run.html#using-the-abi-with-ataddress)
+:::
+
+## Step 5: Set up web3.js and connect to the Ganache network
+
+In this step, we will set up the web3.js library and connect to the Ganache network. So, be sure to run Ganache if you did not already did.
+
+First, install the `web3` package using npm:
+
+```
+npm install web3@4.0.1-rc.1
+```
+
+Note that we are installing the latest version of 4.x, at the time of this tutorial writing. You can check the latest version at https://www.npmjs.com/package/web3?activeTab=versions
+
+Next, create a new file called `index.js` in your project directory and add the following code to it:
+
+```javascript
+const { Web3 } = require('web3');
```

Comment:
I think also mention in desc that we also support ESM imports as we have native ESM builds.

Reply:
Seems a good idea. I will add a hint for this.